test/index.coffee

```diff
@@ -0,0 +1,89 @@
+Promise = require 'bluebird'
+{ expect } = require 'chai'
+_ = require 'lodash'
+
+gc = require('../lib/index')
+dockerUtils = require('../lib/docker.coffee')
+
+SKIP_GC_TEST = process.env.SKIP_GC_TEST == '1' || false
+
+promiseToBool = (p) ->
+	p.return(true).catchReturn(false)
+
+pullAsync = (docker, tag) ->
+	new Promise (resolve, reject) ->
+		docker.pull tag, (err, stream) ->
```

Comment:
This was based upon when you have `_.noop` in your code, it's more about the promise usage (and to make sure you have an error handler on the stream, otherwise it could crash node)

Besides that for your draining problem if you use `stream.resume()` you should be good to drop the `.on 'data'`